agent/grpc/public/services/connectca/sign.go

```diff
@@ -48,6 +48,7 @@ func (s *Server) Sign(ctx context.Context, req *pbconnectca.SignRequest) (*pbcon
 	var rsp *pbconnectca.SignResponse
 	handled, err := s.ForwardRPC(&rpcInfo, func(conn *grpc.ClientConn) error {
 		logger.Trace("forwarding RPC")
+		ctx := public.ForwardMetadataContext(ctx)
```

Comment:
This is critical to have the ACL token (and any other metadata from the incoming request) forwarded. This is due to the gRPC library using different keys to store the incoming and outgoing requests metadata.

Reply:
Wow, good catch! I wonder if there's a way to make this happen with an interceptor, that wouldn't risk leaking tokens to any other gRPC APIs we integrate with in the future 🤔